GooglePlayInstant/Tests/Editor/QuickDeploy/OAuth2ServerTest.cs

```diff
@@ -0,0 +1,107 @@
+
+        private const string AddressPrefix = "http://localhost:5000/";
+
+        [Test]
+        public void TestGetAuthorizationResponse()
+        {
+            var someString = Path.GetRandomFileName();
+            // response is code
+            var codeResponse =
+                OAuth2Server.GetAuthorizationResponse(new Uri(string.Format("{0}?code={1}", AddressPrefix,
+                    someString)));
+            Assert.AreEqual(new KeyValuePair<string, string>("code", someString), codeResponse,
+                "Expected valid code response");
+
+            // response is error
+            var errorResponse =
+                OAuth2Server.GetAuthorizationResponse(new Uri(string.Format("{0}?error={1}", AddressPrefix,
+                    someString)));
+            Assert.AreEqual(new KeyValuePair<string, string>("error", someString), errorResponse,
+                "Expected invalid error response");
+
+            // respose is invalid
+            var invalidUrl = new Uri(string.Format("{0}?someKey=someValue", AddressPrefix));
+            Assert.Throws<ArgumentException>(() => OAuth2Server.GetAuthorizationResponse(invalidUrl),
+                "Expected ArgumentException");
+            LogAssert.Expect(LogType.Error, "Uri query contains invalid params");
+        }
+
+        [Test]
+        public void TestUriContainsValidParams()
+        {
+            const string codeOrErrorKeyMustBePresentText =
+                "Uri query must include either \"code\" or \"error\" as keys.";
+
+            var validUriWithCode = new Uri(string.Format("{0}?code=someValue", AddressPrefix));
+            Assert.IsTrue(OAuth2Server.UriContainsValidQueryParams(validUriWithCode),
+                codeOrErrorKeyMustBePresentText);
+            var uriWithError = new Uri(string.Format("{0}?error=someValue", AddressPrefix));
+            Assert.IsTrue(OAuth2Server.UriContainsValidQueryParams(uriWithError),
+                codeOrErrorKeyMustBePresentText);
+
+
+            const string codeAndErrorCannotBePresentText = "\"code\" and \"error\" cannot be present at the same time.";
+            var invalidUriWithCodeAndError =
+                new Uri(string.Format("{0}?code=codeValue&error=errorValue", AddressPrefix));
+            Assert.IsFalse(OAuth2Server.UriContainsValidQueryParams(invalidUriWithCodeAndError),
+                codeAndErrorCannotBePresentText);
+
+            const string noOtherKeysCanBePresentText =
+                "No other keys apart from \"code\" and \"error\" are allowed.";
+            var invalidUriWithOtherKeys =
+                new Uri(string.Format("{0}?code=codeValue&otherKey=someValue", AddressPrefix));
+            Assert.IsFalse(OAuth2Server.UriContainsValidQueryParams(invalidUriWithOtherKeys),
+                noOtherKeysCanBePresentText);
+        }
+
+        [Test]
+        public void TestGetQueryParamsFromUri()
+        {
+            var testDict = new Dictionary<string, string>();
```

Comment:
Please change to something simpler like this:

var expected = new Dictionary<string, string> {{"a", "b"}, {"c", "d"}};
var uri = new Uri(AddressPrefix + "?a=b&c=d");

Also test boundary conditions like null and empty